src/main/java/com/imsweb/mph/DefaultHematoDbUtilsProvider.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2016 Information Management Services, Inc.
+ */
+package com.imsweb.mph;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import au.com.bytecode.opencsv.CSVReader;
+
+public class DefaultHematoDbUtilsProvider implements HematoDbUtilsProvider {
+
+    private static Map<String, List<HematoDbDTO>> _SAME_PRIMARY_DTO = new HashMap<>();
+    private static Map<String, List<HematoDbDTO>> _TRANSFORM_TO_DTO = new HashMap<>();
+    private static Map<String, List<HematoDbDTO>> _TRANSFORM_FROM_DTO = new HashMap<>();
+    private static Pattern _MORPHOLOGY = Pattern.compile("^(\\d{4}/\\d)");
+
+    private DefaultHematoDbUtilsProvider() {
+        initializeLookup();
+    }
+
+    public static DefaultHematoDbUtilsProvider getInstance() {
+        return new DefaultHematoDbUtilsProvider();
+    }
+
+    public boolean isSamePrimary(String leftCode, String rightCode, int year) {
+        if (leftCode == null || rightCode == null || !_MORPHOLOGY.matcher(leftCode).matches() || !_MORPHOLOGY.matcher(rightCode).matches())
+            return false;
+        else if (leftCode.equals(rightCode))
+            return true;
+        else if (_SAME_PRIMARY_DTO.containsKey(leftCode)) {
+            for (HematoDbDTO dto : _SAME_PRIMARY_DTO.get(leftCode))
+                if (dto.matches(rightCode, year))
+                    return true;
+        }
+        else if (_SAME_PRIMARY_DTO.containsKey(rightCode)) {
+            for (HematoDbDTO dto : _SAME_PRIMARY_DTO.get(rightCode))
+                if (dto.matches(leftCode, year))
+                    return true;
+        }
+        return false;
+    }
+
+    public boolean isAcuteTransformation(String leftCode, String rightCode, int year) {
+        if (leftCode == null || rightCode == null || !_MORPHOLOGY.matcher(leftCode).matches() || !_MORPHOLOGY.matcher(rightCode).matches())
+            return false;
+        else if (_TRANSFORM_TO_DTO.containsKey(leftCode)) {
+            for (HematoDbDTO dto : _TRANSFORM_TO_DTO.get(leftCode))
+                if (dto.matches(rightCode, year))
+                    return true;
+        }
+        return false;
+    }
+
+    public boolean isChronicTransformation(String leftCode, String rightCode, int year) {
+        if (leftCode == null || rightCode == null || !_MORPHOLOGY.matcher(leftCode).matches() || !_MORPHOLOGY.matcher(rightCode).matches())
+            return false;
+        else if (_TRANSFORM_FROM_DTO.containsKey(leftCode)) {
+            for (HematoDbDTO dto : _TRANSFORM_FROM_DTO.get(leftCode))
+                if (dto.matches(rightCode, year))
+                    return true;
+        }
+        return false;
+    }
+
+    private static synchronized void initializeLookup() {
```

Comment:
If you make the constructor public and remove the getInstance(), you don't need this method, you can just put the logic in the constructor. Those big collections on the top of the class don't need to be static either...